### PR TITLE
Window::set_minimized (#985)

### PR DESCRIPTION
**Pull request by SuperiorJT**
Gets started on supporting `Window::set_minimized` (#985, #983). I'm very new to this repo and a beginner in Rust so it might not be exactly how you guys want it. I've tested it on macOS via keyboard events and was able to make it minimize similar to other macOS applications. I'm willing to work on the Windows side of this as well.

- [x] Tested on all platforms changed
  - [x] Windows
  - [x] macOS
  - [x] X11
  - [x] Wayland
  - [x] iOS
- [x] `cargo fmt` has been run on this branch
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/platform_impl/linux/wayland/window.rs [hunk 1]
```diff
@@ -278,6 +278,10 @@ impl Window {
         *(self.need_frame_refresh.lock().unwrap()) = true;
     }
 
+    pub fn set_minimized(&self, minimized: bool) {
+        unimplemented!();
```
Comment (vberger) [inline src/platform_impl/linux/wayland/window.rs]:
```suggestion
        // An app cannot un-minimize itself on Wayland
        if minimized {
            self.frame.lock().unwrap().set_minimized();
        }
```

**Comment by SuperiorJT** [comment]
On the Windows implementation do we want to create a `WindowFlag` for `MINIMIZED`? I'm looking at the way `set_maximized` works and it looks like we could do something similar, but I don't know if we should track that in our `WindowState`.

**Comment by Osspial** [comment]
@SuperiorJT We should track that in `WindowFlag`s, yeah.

**Comment by ZeGentzy** [comment]
@SuperiorJT What is the status on this PR?

**Comment by SuperiorJT** [comment]
Implementations are made for windows, macOS and wayland. We need someone to develop the x11 and iOS (if applicable) implementation since I am unable to. Once all is implemented, then I can go through and fmt and update required documentation.

**Comment by murarth** [comment]
I have an implementation for X11 ready: https://github.com/murarth/winit/tree/x11-set-minimize

**Comment by SuperiorJT** [comment]
I added @murarth's X11 implementation, formatted, and added documentation. It's ready for review.

File: src/platform_impl/android/mod.rs [hunk 2]
```diff
@@ -359,6 +359,11 @@ impl Window {
         Err(ExternalError::NotSupported(NotSupportedError::new()))
     }
 
+    #[inline]
+    pub fn set_minimized(&self, _minimized: bool) {
+        unimplemented!()
+    }
```
Comment (murarth) [inline src/platform_impl/android/mod.rs]: Do we want this to panic or should it be a no-op? `set_maximized` below is a no-op.

Comment (Osspial) [reply src/platform_impl/android/mod.rs]: I don't think it particular matters, since the Android backend doesn't even compile right now.

**Comment by Osspial** [comment]
I've tested this out on Windows, and this seems to cause glitches if you do the following:

1. Maximize the window
2. Minimize the window via `set_minimized`
3. Alt-tab back into the window

If you do that, the minimize button in the window's titlebar has the window restore icon instead of the window minimize icon, and fails to minimize the window when pressed.

**Comment by SuperiorJT** [comment]
I've fixed the maximize issue. Windows didn't like the window styles being set after minimization. Let me know if you find any more issues.

**Comment by murarth** [comment]
@goddessfreya I wrote the X11 implementation myself, so I can confirm it works. If we want to ping our X11 testers to get another set of eyes on it, that's fine by me.

**Comment by goddessfreya** [comment]
@murarth Haha, I'm ever so slightly blind. I see. Did you test for weird behavior evolving fullscreened/unmapped windows? Should be good to go on the X11 side, I don't see much that could go wrong.

**Comment by murarth** [comment]
@goddessfreya I've tested for any weird interactions and found none.

**Comment by SuperiorJT** [comment]
I added the stub for WASM. Let me know if I missed anything.

**Pull request merged**